tests/gatewayapitests/utils.go

```diff
@@ -343,6 +358,21 @@ func GetParentReferencesV1(gatewayNames []string, namespace string, ports []int3
 	return parentRefs
 }
 
+func GetParentReferencesOnListener(listeners []gatewayv1.Listener, gwName, namespace string) []gatewayv1.ParentReference {
```

Comment:
Can this function be renamed ?
May be "GetParentReferencesForListener/GetParentReferencesPerListener/Anything else that you may like which says it returns parentReferences corresponding to listener"

Reply:
done